skada/_ot.py

```diff
@@ -0,0 +1,185 @@
+
+        # sole OT problem
+        sol = ot.solve(M)
+
+        T = sol.plan
+        loss_ot = sol.value
+
+        if i == 0:
+            loss_ot += alpha * np.sum(Ml * T)
+
+        lst_loss_ot.append(loss_ot)
+
+        # compute the transported labels
+        yth = nt * ys.T.dot(T)
+
+        # fit the estimator
+        estimator.fit(Xt, yth, **kwargs)
+        y_pred = estimator.predict(Xt)
+
+        Ml = ot.dist(ys.reshape(-1, 1), y_pred.reshape(-1, 1))
+
+        # compute the loss
+        loss_tgt_labels = np.mean((yth - y_pred)**2)
+        lst_loss_tgt_labels.append(loss_tgt_labels)
+
+        if verbose:
+            print(f'iter={i}, loss_ot={loss_ot}, loss_tgt_labels={loss_tgt_labels}')
+
+        # break on tol OT loss
+        if i > 0 and abs(lst_loss_ot[-1] - lst_loss_ot[-2]) < tol:
+            break
+
+        # break on tol target loss
+        if i > 0 and abs(lst_loss_tgt_labels[-1] - lst_loss_tgt_labels[-2]) < tol:
+            break
+
+        # update the cost matrix
+        if i == n_iter_max - 1:
+            warnings.warn('Maximum number of iterations reached.')
+
+    return estimator, lst_loss_ot, lst_loss_tgt_labels, sol
+
+
+class JDOTRegressor(DAEstimator):
+    """Joint Distribution Optimal Transport Regressor
+
+    Parameters
+    ----------
+    base_estimator : object
+        The base estimator to be used for the regression task. This estimator
+        should solve a least squares regression problem (regularized or not)
+        to correspond to JDOT theoretical regression problem but other
+        approaches can be used with the risk that the fixed point might not converge.
+    alpha : float, default=0.5
+        The trade-off parameter between the feature and label loss in OT metric
+    n_iter_max: int
+        Max number of JDOT alternat optimization iterations.
+    tol: float>0
+        Tolerance for loss variations (OT and mse) stopping iterations.
+    verbose: bool
+        Print loss along iterations if True.as_integer_ratio
+
+    Attributes
+    ----------
+    estimator_ : object
+        The fitted estimator.
+    lst_loss_ot_ : list
+        The list of OT losses at each iteration.
+    lst_loss_tgt_labels_ : list
+        The list of target labels losses at each iteration.
+    sol_ : object
+        The solution of the OT problem.
+
+    References
+    ----------
+    [1] N. Courty, R. Flamary, A. Habrard, A. Rakotomamonjy, Joint Distribution
+        Optimal Transportation for Domain Adaptation, Neural Information
+        Processing Systems (NIPS), 2017.
+
+    """
+
+    def __init__(self, base_estimator, alpha=0.5, n_iter_max=100,
+                 tol=1e-5, verbose=False, **kwargs):
+        self.base_estimator = base_estimator
+        self.kwargs = kwargs
+        self.alpha = alpha
+        self.n_iter_max = n_iter_max
+        self.tol = tol
+        self.verbose = verbose
+
+    def fit(self, X, y=None, sample_domain=None, *, sample_weight=None):
```

Comment:
With the routing mechanism, I guess there's no need to specify `sample_weight` in the list of arguments if don't use those.

Reply:
I do not use it yet but I wanted to implement with weights in next commits, Thanks for reminding me.

Reply:
Done I needed to update `split_domain` function to handle None samples_weight